### PR TITLE
[STREAMING][MINOR] More contextual information in logs + minor code i…

### PR DESCRIPTION
**Pull request by jaceklaskowski**
…mprovements

Please review and merge at your convenience. Thanks!

### CODE REVIEW

**Comment by rxin** [comment]
Can you combine your pull requests into a single one?

**Comment by jaceklaskowski** [comment]
Yes, sure! Been worried that @srowen might cross few lines out that would completely devastate my mood today :) On to...

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala [hunk 1]
```diff
@@ -59,17 +59,15 @@ case class JobSet(
 
   // Time taken to process all the jobs from the time they were submitted
   // (i.e. including the time they wait in the streaming scheduler queue)
-  def totalDelay: Long = {
-    processingEndTime - time.milliseconds
-  }
+  def totalDelay: Long = processingEndTime - time.milliseconds
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala]: Although I wouldn't bother with this kind of change, it's OK here IMHO

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala [hunk 2]
```diff
@@ -59,17 +59,15 @@ case class JobSet(
...
 
   def toBatchInfo: BatchInfo = {
     BatchInfo(
       time,
       streamIdToInputInfo,
       submissionTime,
-      if (processingStartTime >= 0) Some(processingStartTime) else None,
-      if (processingEndTime >= 0) Some(processingEndTime) else None,
+      if (hasStarted) Some(processingStartTime) else None,
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala]: These change the logic slightly -- are you sure it's equivalent?

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 3]
```diff
@@ -286,7 +286,7 @@ abstract class DStream[T: ClassTag] (
     dependencies.foreach(_.validateAtStart())
 
     logInfo("Slide time = " + slideDuration)
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: It'd be fine to make this all use interpolation while you're at it

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala (hunk 1 above)
Comment (jaceklaskowski) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala]: Noted & thanks!

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala (hunk 2 above)
Comment (jaceklaskowski) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala]: Tested it locally (and can't wait to see the results from Jenkins).

The current code *overly* assumes that the times can be `0` (which cannot...ever). It is also more clearer that at `hasCompleted` `processingEndTime` is already set. It's over-complicated as it's now IMHO.

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala (hunk 3 above)
Comment (jaceklaskowski) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: Thanks! I was thinking about it, but was worried to propose such changes as you might not like it :)

**Comment by jaceklaskowski** [comment]
Merged the other branches and ran build locally. Please review and merge at your convenience @srowen @rxin. Thanks!

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
OK, I see you replaced a lot more with interpolation. I guess that's OK. But I think it's why you have a merge conflict now that needs to be resolved.

**Comment by jaceklaskowski** [comment]
On it. I'll rebase and push update. Thanks @srowen for code review.

**Comment by jaceklaskowski** [comment]
Changes rebased and ready for review.

**Comment by srowen** [comment]
Jenkins, test this please

**Comment by srowen** [comment]
Jenkins, test this please

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**